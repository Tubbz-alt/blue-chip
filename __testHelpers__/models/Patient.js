import BaseModel from "../../src/BaseModel";
import SettingOfUse from "./SettingOfUse";
import Indication from "./Indication";

export default class Patient extends BaseModel {
  static get belongsTo() {
    return [SettingOfUse, Indication];
  }
}
