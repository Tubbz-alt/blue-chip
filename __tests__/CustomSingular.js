import resources from "../__testHelpers__/fixtures/patientsAndSettingsOfUseNormalized";
import Patient from "../__testHelpers__/models/Patient";

describe("Custom singular", () => {

    test("includes settingOfUse singular for belongs to", () => {
        const patients = Patient.query(resources)
            .all()
            .includes(["settingOfUse"])
            .toObjects();
        expect(patients).toMatchSnapshot();
    });

});
