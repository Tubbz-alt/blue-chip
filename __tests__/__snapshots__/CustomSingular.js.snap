// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom singular includes settingOfUse singular for belongs to 1`] = `
Array [
  Object {
    "createdAt": "2018-10-08T18:39:33.000Z",
    "customers": Array [
      "0000",
    ],
    "id": "667",
    "lastSessionDate": "2016-10-01T00:00:00.000Z",
    "myPatient": true,
    "patientId": "0000",
    "serialNumbers": Array [
      "21601009",
    ],
    "sessionsCount": 127001,
    "settingOfUse": Object {
      "id": "1",
      "name": "Inpatient",
    },
    "steps": 24099980,
    "upTime": 68599053,
  },
]
`;
